extensions/ql-vscode/test/vscode-tests/no-workspace/data-extensions-editor/auto-model-codeml-queries.test.ts

```diff
@@ -0,0 +1,188 @@
+import { createMockLogger } from "../../../__mocks__/loggerMock";
+import type { Uri } from "vscode";
+import {
+  DatabaseItem,
+  DatabaseKind,
+} from "../../../../src/databases/local-databases";
+import { file } from "tmp-promise";
+import { QueryResultType } from "../../../../src/query-server/new-messages";
+import { runAutoModelQueries } from "../../../../src/data-extensions-editor/auto-model-codeml-queries";
+import { Mode } from "../../../../src/data-extensions-editor/shared/mode";
+import { mockedObject } from "../../utils/mocking.helpers";
+import { CodeQLCliServer } from "../../../../src/codeql-cli/cli";
+import { QueryRunner } from "../../../../src/query-server";
+import * as queryResolver from "../../../../src/local-queries/query-resolver";
+import * as standardQueries from "../../../../src/local-queries/standard-queries";
+
+function createMockUri(path = "/a/b/c/foo"): Uri {
```

Comment:
Would be great to move this to a file (like other mocks/stubs) so that it can be re-used easily.